base/iobuffer.jl

```diff
@@ -32,15 +32,27 @@ StringVector(n::Integer) = unsafe_wrap(Vector{UInt8}, _string_n(n))
 # IOBuffers behave like Files. They are typically readable and writable. They are seekable. (They can be appendable).
 
 """
-    IOBuffer([data, ][readable::Bool=true, writable::Bool=false[, maxsize::Int=typemax(Int)]])
+    IOBuffer([data::AbstractVector{UInt8}]; keywords...) -> IOBuffer
 
-Create an `IOBuffer`, which may optionally operate on a pre-existing array. If the
-readable/writable arguments are given, they restrict whether or not the buffer may be read
-from or written to respectively. The last argument optionally specifies a size beyond which
-the buffer may not be grown.
+Create an in-memory I/O stream, which may optionally operate on a pre-existing array.
+
+It may take optional keyword arguments:
+- `read`, `write`, `append`: restricts operations to the buffer; see `open` for details.
+- `truncate`: truncates the buffer size to zero length.
+- `maxsize`: specifies a size beyond which the buffer may not be grown.
+
+When `data` is given, the buffer will be both readable and writable by default.
```

Comment:
I don't think this is true --- the keyword arguments default to `nothing, which `open_flags` interprets as read-only. And that's probably a good thing; for example we have

```
IOBuffer(str::String) = IOBuffer(unsafe_wrap(Vector{UInt8}, str))
```

which we want to remain read-only.

Reply:
You are right. I meant "When `data` is not given, ...". Fixed.